iceoryx_hoofs/include/iceoryx_hoofs/cxx/unique_ptr.hpp

```diff
@@ -70,11 +68,6 @@ class unique_ptr
     ///
     const T* operator->() const noexcept;
 
-    ///
-    /// @brief operator bool Returns true if it points to something.
-    ///
-    explicit operator bool() const noexcept;
```

Comment:
It will not be possible anymore to find out if a `cxx::unique_ptr` is in a moved from object by using `operator bool()`. Not sure if we should force the user to call `get()` to check :thinking:

Reply:
No. The compiler will warn you if you do this. This is the use after move warning we are explicitly suppressing in our tests.

Reply:
@elfenpiff Compiler? You mean clang-tidy?

Reply:
No I mean gcc>=12, clang>=14 and of course also clang-tidy